src/IssueViz.Security.UnitTests/Commands/TaintToolWindowCommandTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Design;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.IssueVisualization.Security.Commands;
+using SonarLint.VisualStudio.IssueVisualization.Security.UI.TaintList;
+using Constants = SonarLint.VisualStudio.IssueVisualization.Security.Commands.Constants;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.UnitTests.Commands
+{
+    [TestClass]
+    public class TaintToolWindowCommandTests
```

Comment:
Copy of the corresponding hotspots tests.

Looking at the implementation of the command class, I think it should be possible to refactor it to a single general purpose "show tool window" command implementation that is just passed the [command guid+id] and [toolwindow guid]. I'll look that separately.

The `IssueVisualizationToolWindowCommand` is similar but slightly more complicated (it registers two commands and sets the enabled status based an active UI context). I'll ignore it for now, but it's something we might want to add to the security "show tool window" commands in the future e.g. the show hotspots window command only be enabled if there is a hotspot in the list.

Reply:
I agree, good idea. The boilerplate is identical.